src/compile.rs

```diff
@@ -329,6 +344,84 @@ pub fn build_func_map<'a>(
     Ok(decl_map)
 }
 
+pub fn validate_fs_context_duplicates(
+    statements: &BTreeSet<ValidatedStatement>,
+) -> Result<(), CascadeErrors> {
+    let mut errors = CascadeErrors::new();
+    let mut fsc_rules: BTreeMap<&String, BTreeSet<&FileSystemContextRule>> = BTreeMap::new();
+
+    for statement in statements {
```

Comment:
I feel like this for loop could live at the validate_rules().  With it in here, we'll need to loop over the entire AST for each validation check we eventually add.  If we move it to the validate_rules() level and then pass fsc_rules to validate_fs_context_duplicates, then we can set up the data everyone needs in a single AST pass at the top level of validate_rules().